pkg/controller/servicebinding/secret.go

```diff
@@ -27,17 +34,46 @@ func (s *secret) buildResourceClient() dynamic.ResourceInterface {
 	return s.client.Resource(gvr).Namespace(s.ns)
 }
 
-// createOrUpdate will take informed payload and either create a new secret or update an existing
-// one. It can return error when Kubernetes client does.
-func (s *secret) createOrUpdate(payload map[string][]byte, ownerReference metav1.OwnerReference) (*unstructured.Unstructured, error) {
+// HashedResourceNameForData is a utility function to get a checksum of the resource data and append it to resource name
+func HashedResourceNameForData(name string, data map[string][]byte) (string, error) {
```

Comment:
@isutton Asked me to change the name of the function... Please refer to the older conversation. 
I cannot keep making the same changes again and again.

Reply:
The secret type doesn't represent a single resource anymore. It should be called as a single function now that all the parameters are being set at its creation. 

So instead of creating the secret and passing it further, change the call site to create the secret there, since the only method we're interested is createOrUpdate or some incarnation of it.

Reply:
> Asked me to change the name of the function... Please refer to the older conversation.
I cannot keep making the same changes again and again.

@DhritiShikhar, this is the discussion you are referring https://github.com/redhat-developer/service-binding-operator/pull/750#discussion_r546731896 ?

I do not see as doing the same change again and again - it is rather an iterative improvement, given that now we have all needed data available in `secret` struct, we can see that `HashedResourceNameForData` is actually computing the secret name and hence we should name it like this, for better code readability. BTW, I could not see that https://github.com/redhat-developer/service-binding-operator/pull/750#discussion_r536155255 were addressed in PR, for example `name()` function does not exist.

Reply:
Done